torchao/prototype/quantization/codebook/test_codebook_quant.py

```diff
@@ -0,0 +1,70 @@
+import unittest
+
+import torch
+
+from torchao.prototype.quantization.codebook import (
+    CodebookQuantizedTensor,
+    choose_qparams_codebook,
+)
+
+
+class TestCodebookQuantization(unittest.TestCase):
+    def setUp(self):
+        torch.manual_seed(123)
+        self.input = torch.randn(100, 256, dtype=torch.float32)
+        self.block_size = (1, 1)
+        self.scale_block_size = 64
+        self.code_dtype = torch.uint4
+        self.chunk_size = 1024
+
+    def test_choose_qparams_codebook(self):
+        codebook, scales = choose_qparams_codebook(
+            self.input,
+            block_size=self.block_size,
+            scale_block_size=self.scale_block_size,
+            code_dtype=self.code_dtype,
+        )
+        self.assertEqual(codebook.dim(), len(self.block_size) + 1)
+
+        self.assertFalse(torch.isnan(codebook).any())
+        self.assertFalse(torch.isnan(scales).any())
+
+    def test_codebook_quantized_tensor_from_float(self):
+        cqt = CodebookQuantizedTensor.from_float(
+            self.input,
+            block_size=self.block_size,
+            code_dtype=self.code_dtype,
+            scale_block_size=self.scale_block_size,
+            chunk_size=self.chunk_size,
+        )
+
+        dequant = cqt.dequantize()
+
+        torch.testing.assert_close(dequant, self.input, atol=2, rtol=2)
```

Comment:
looks like a large drop, we could use a larger dtype for test I think, so we can get something closer